students/brgalloway/Lesson_3/mailroom_part_1.py

```diff
@@ -15,26 +16,48 @@
     ["Paul Allen", 708.42, 3, 236.14],
     ["William Gates, III", 653784.49, 2, 326892.24],
     ["Bill Ackman", 2354.05, 3, 784.68],
-    ["Mark Zuckerberg", 16396.10, 3, 5465.37]  
+    ["Mark Zuckerberg", 16396.10, 3, 5465.37] 
 ]
 
 # Main function to get users input
-def user_prompt(donors_list):
+def main_prompt():
     display_menu = "Choose one of the following options. \n\n" \
                 "1 - Send a Thank You \n" \
                 "2 - Create a Report \n" \
                 "3 - Quit \n"
-    print(display_menu)
-    prompt = input("Enter a choice to continue: ")   
     while True:
+        print(display_menu)
+        prompt = input("Enter a choice to continue: ")  
         if prompt == "1":
-            find_donor()
-            break
+            sub_menu()
         elif prompt == "2":
             generate_report(donors_list)
-            break
-        else: 
-            break
+        elif prompt == "3":
+            sys.exit()
+        else:
+            print("Enter a valid response")
+
+def sub_menu():
+    while True:
+        fullname = input("Enter the full name of the donor or type list to display names\n" \
+                        "Typing quit will take you to the main menu\n" \
+                        ">> ")
+        if fullname == "list":
+            return list_names()
+        elif fullname == "quit":
+            return main_prompt()
```

Comment:
I would avoid creating sub-loops as it can create weird behaviour. Since you're still in a while True loop in your main loop you can just do `return` without executing `main_prompt()` again and creating a secondary while True loop in that function.